tests/Spec.hs

```diff
@@ -0,0 +1,79 @@
+{-# LANGUAGE FlexibleContexts #-}
+module Main where
+
+import Prelude as P
+import Conduit
+import Control.Monad (void, when, zipWithM_)
+import Test.Hspec
+import Data.Conduit.Tar
+import System.Directory
+import Data.ByteString as S
+import System.IO
+import System.FilePath
+import Control.Exception
+
+main :: IO ()
+main = do
+    let baseTmp = "tar-conduit-tests"
+    isStack <- doesDirectoryExist ".stack-work"
+    let testPaths =
+            ["src", "./tests", "README.md", "ChangeLog.md", "LICENSE"] ++
+            if isStack
+                then [".stack-work", "./sample"]
+                else []
+    hspec $ do
+        describe "tar/untar" $ do
+            let tarUntarContent dir =
+                    runConduitRes $
+                    yield dir .| void tarFilePath .| untar (const (foldC >>= yield)) .| foldC
+            before (collectContent "src") $ it "content" (tarUntarContent "src" `shouldReturn`)
```

Comment:
Why bother with putting `collectContent` into `before` instead of just running it inside the `it` block?

Reply:
`collectContent` creates the expected value and isn't really part of the test itself, at least maybe that is what my thought process was at the time. There is no real difference here, so I'll just put it inside of `it`, if you think it's cleaner that way.

Reply:
:man_shrugging: Not really sure. I'd just never seen the `before` approach... before. Your call.